pkg/apis/serving/v1alpha2/util.go

```diff
@@ -0,0 +1,87 @@
+package v1alpha2
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	"k8s.io/kubernetes/pkg/apis/core"
+	"k8s.io/kubernetes/pkg/apis/core/validation"
+)
+
+func setResourceRequirementDefaults(requirements *v1.ResourceRequirements) {
+	if requirements.Requests == nil {
```

Comment:
Consider implementing with a for loop which makes it easier to extend this in the future (accelerators?). More importantly, it guarantees that the logic is the same and doesn't have errors in different cases.

for value, key := map[key]value{
  v1.ResourceCPU: DefaultCPU,
  v1.ResourceMemory: DefaultMemory
} {
...
}

Reply:
It's existing code and there's definition in predictor.go:
```
var (
	DefaultMemory = resource.MustParse("2Gi")
	DefaultCPU    = resource.MustParse("1")
)
```
I can move it to the util.go(maybe constants?) and refactor it to be loop a map.

Reply:
Logged issue #496 for this.